routes/users.py

```diff
@@ -48,3 +48,17 @@ async def delete_user(identifier: models.UserIdentifier):
 async def get_user(identifier: models.UserIdentifier):
     user_data = await utils.get_user_info_by_identifier(identifier)
     return models.UserInfoQueryResponse(**user_data.dict())
+
+
+# this implementation doesn't use models.ValidateUserResponse. Should I implement this?
+# also, how can I define a 404 response if it user isn't found
+@router.post("/users/validate",  # todo: finish this endpoint
+             response_model=models.ValidateLoginResponse,
+             description=docs.validate_user_desc,
+             summary=docs.validate_user_summ,
+             tags=["Users"],
+             status_code=200)
+async def validate_user(validate: models.ValidateUserForm):  # todo: figure out how this works
+    return await utils.validate_user_info(validate)
```

Comment:
This code looks perfect. 

But here's a broader design question to worry about: is `validate` the best way to call this flow? 
In a micro sense, we are just validating a user by their password, but that doesn't really mean much. Does `validate_user` validate the user's data? Does it validate that the user _exists_? 

If what we're doing is validating the user's password, then we should use the common name for that: `user_login`.

This makes it instantly obvious to anyone who needs to read the code, as well as being a universal workflow that _anyone_ can understand.

While it might seems annoying and unnecessary, small changes like these keep the code clean and healthy.

Also, no one wants to be the one to have to explain to frontend why their login endpoint isn't called login hahaha.

There are other comments mimicking this one in the modules that also need name changes, use them as reminders for what code needs to be fixed.